server/src/main/java/io/crate/planner/statement/CopyFromPlan.java

```diff
@@ -288,28 +308,81 @@ public static ExecutionPlan planCopyFromExecution(AnalyzedCopyFrom copyFrom,
                 lineNumberSymbol
             );
         } else {
-            sourceIndexWriterProjection = new SourceIndexWriterProjection(
+
+            // Add top-level columns (filter if we know already target columns) in addition to required sub-columns added above.
+            // If required columns are top-level columns, they are already added and ignored here.
+            table.columns()
+                .stream()
+                .filter(ref ->
+                    // if targetColumns are empty CSV header/json keys are source of the truth and plan has to be adjusted in runtime to
+                    // either dynamically add columns or remove table columns, not listed in header.
+                    boundedCopyFrom.targetColumns().isEmpty()
+                        ||
+                        // Regular filtering
+                        (boundedCopyFrom.targetColumns().isEmpty() == false && boundedCopyFrom.targetColumns().contains(ref.column().sqlFqn()))
+                )
+                .forEach(ref -> {
+                    boolean added = targetColumns.add(ref);
+                    if (added) {
+                        toCollect.add(ref);
+                    }
+                });
+
+            List<Reference> targetColsExclPartitionCols = new ArrayList<>();
+            for (Reference column : targetColumns) {
+                if (table.partitionedBy().contains(column.column())) {
+                    continue;
+                }
+                targetColsExclPartitionCols.add(column);
+            }
+
+            // For generated columns, we first try to get provided in a file value and validate it.
+            // If it's not provided we generate it ourselves.
+            for (int i = 0; i < toCollect.size(); i++) {
+                Symbol symbol = toCollect.get(i);
+                if (symbol instanceof GeneratedReference genRef) {
+                    var function = new io.crate.expression.symbol.Function(
+                        CoalesceFunction.SIGNATURE,
+                        List.of(genRef, genRef.generatedExpression()),
+                        Symbols.typeView(List.of(genRef.generatedExpression())).get(0)
+                    );
+                    toCollect.set(i, function);
+                    coalesceFunctionsByGenRef.put(genRef.column(), function);
+                }
+            }
+
+            sourceSymbols = new InputColumns.SourceSymbols(toCollect);
+
+            if (clusteredBy != null) {
+                clusteredByInputCol = inputColumnsForReference(table.getReference(clusteredBy), coalesceFunctionsByGenRef, sourceSymbols);
+            }
+
+            indexWriterProjection = new ColumnIndexWriterProjection(
                 table.ident(),
                 partitionIdent,
```

Comment:
BTW, on current master we always set `partitionIdent` to null in the only usage of this ctor in `InsertFromSubQueryPlanner`.

However, I don't think it's related to flakiness of 5.3 insert into https://github.com/crate/crate/pull/14232

Here we set partitionIdent but similar `testCopyFromToPartitionedTableWithNullValue` is still flaky.